src/components/hack-details.js

```diff
@@ -1,100 +1,174 @@
-      );
-      setGoal("To win and get that dosh but also the experience");
+      Axios.get(
+        process.env.REACT_APP_API_URL + "hackdetail/" + props.match.params.id
+      ).then(res => {
+        setTitle(res.data.title);
+        setDescription(res.data.description);
+        setGoal(res.data.goal);
+        setTeam(res.data.team);
+      });
     }
   }, [data, props.match]);
 
-  return (
-    <Grid
-      container
-      direction="column"
-      justify="center"
-      alignItems="stretch"
-      alignContent="center"
-    >
-      <Paper className={classes.root}>
-        <Grid
-          container
-          justify="space-between"
-          alignItems="center"
-          alignContent="center"
-          spacing={1}
-        >
-          <Grid item xs={10}>
-            <Chip
-              icon={<ErrorOutlineOutlinedIcon />}
-              label="status"
-              color="secondary"
-            />
-          </Grid>
-          <Grid item xs={2} className={classes.rightFeild}>
-            <Button variant="contained" color="primary">
-              Join Hack
-            </Button>
-          </Grid>
-          <Grid item xs={12}>
-            <Typography variant="h4">{title}</Typography>
-          </Grid>
-          <Grid item xs={12}>
-            <Typography variant="h6">Idea:</Typography>
-          </Grid>
-          <Grid item xs={10}>
-            <Typography variant="body1">{description}</Typography>
-          </Grid>
-          <Grid item xs={12}>
-            <Typography variant="h6">Goal:</Typography>
-          </Grid>
-          <Grid item xs={10}>
-            <Typography variant="body1">{goal}</Typography>
-          </Grid>
-          <Grid item xs={12}>
-            <Typography variant="h6">Team:</Typography>
+  const joinHack = event => {
+    const object = {
+      hackId: props.match.params.id,
+      userId: props.userId
+    };
+    Axios.post(process.env.REACT_APP_API_URL + "joinhack", object).then(res => {
+      setTeam(res.data.team);
+    });
+  };
+
+  const joinHackButton = () => {
+    if (team !== undefined) {
+      if (team.includes(props.userId)) {
+        return (
+          <Button
+            variant="contained"
+            color="primary"
+            onClick={joinHack}
+            disabled
+          >
+            Join Hack
+          </Button>
+        );
+      } else {
+        return (
+          <Button variant="contained" color="primary" onClick={joinHack}>
+            Join Hack
+          </Button>
+        );
+      }
+    }
+    return (
+      <Button variant="contained" color="primary" onClick={joinHack}>
+        Join Hack
+      </Button>
+    );
+  };
```

Comment:
This whole thing could just be

```
<Button
  variant="contained"
  color="primary"
  onClick={joinHack}
  disabled={team && team.includes(props.userId)}
>
  Join Hack
</Button>
```

...right?

Reply:
... didn't know you could do this

Reply:
This did work except for when the user clicked on the button, however since the button is now its own component this isn't an issues anymore